### DISCUSSION TITLE
Adding tron(trx) support

### DISCUSSION THREAD

**Discussion by Mandofskii** [btcpayserver/btcpayserver#3186] [Ideas / Feature Requests]
Can we add support for trx?
Because all of my costumers use tron for payments and I want add support for trx
Regards

**Comment by IT4Bitcoin** [comment]
Hello people, happy new year.
 I am not a big fan of Tron but adding USDT over Tron it would be very handy, even bitrefill use it nowadays.
 I use USDT over Liquid mainly, but it is not widely accepted yet. Adding support to this we can even automatize swaps between both wich is not widely available (only BTSE and you need to sign up).

**Comment by lucidddreams** [comment]
There's a plugin called SideShift in btcpay, you don't need to complicate things.

I works like this 1) customer opens pay invoice, 2) selects conversion > pay with sideshift (includes tokens/usdt) 3. done.

It works like magic and you don't need to convert multiple cryptos

**Reply by astupidmoose** [reply]
Tron isn't supported on Sideshift.

**Comment by IT4Bitcoin** [comment]
I will pay a bounty of 1 Million sats and release the code to the community if someone writes the alternative plugin for https://fixedfloat.com that supports TRC20 and Lightning ⚡

@Kukks maybe?

**Reply by Kukks** [reply]
![image](https://user-images.githubusercontent.com/1818366/152148380-04e5588f-8659-4852-8fd7-3bd0249d7f08.png)

https://btcpay.kukks.org/payment-requests/f25914f3-bc81-47b1-9ff9-bedd781f2bdc

**Reply by IT4Bitcoin** [reply]
You are awesome buddie.
It is already settled. Thanks!

I will try it in a while and provide feedback

**Reply by IT4Bitcoin** [reply]
I am reporting a Lightning network issue in the commit https://github.com/btcpayserver/btcpayserver-plugins/commit/b1e53ccde39b8d14be8a10d9034b4c85ec7d4c53#commitcomment-65872565
Thanks

USDT (TRC20)>BTC(on-chain) is already working!  
FYI @Serqati, here you have a little DEMO. We can undoubtedly mark the Feature Request as SOLVED (in the best possible way because we did not shitcoined the code)

https://user-images.githubusercontent.com/58536239/152209935-9819e625-6307-47a5-8825-88fc7a3902a8.mp4

.

**Reply by Kukks** [reply]
Can you create an issue tracking all of these problems at https://github.com/btcpayserver/btcpayserver-plugins?

These are all issues from their end. I've already emailed them to report them.

**Reply by IT4Bitcoin** [reply]
Sure, I will keep track on this and talk with Fixed float team. Thanks @Kukks

**Reply by IT4Bitcoin** [reply]
Wow, I see that issue fixed already in the new release. Good job!

**Comment by bolatovumar** [comment]
I have it on good authority that @astupidmoose REALLY wants Tron support.

**Comment by nosovk** [comment]
there is a plugin now: https://github.com/btcpayserver-tether/BTCPayServer.Plugins.USDt
